Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -312,33 +355,44 @@ class InAppMessageViewTest: UnitTest {
+    // Only tells you if the View is visible in the UI to the user. Does not tell you if the View is in the process of rendering a message.
+    func isInlineViewVisible(_ view: InAppMessageView) -> Bool {
+        guard let viewHeightConstraint = view.heightConstraint else {
+            return false
+        }
 
-        // When sizeChanged() is called on the inline View, it adds a task to the main thread queue. Our test wants to wait until this task is done running.
-        await waitForMainThreadToFinishPendingTasks()
+        return viewHeightConstraint.constant > 0
+    }
+
+    // Tells you the message the Inline View is either rendering or has already rendered.
+    func getInAppMessage(forView view: InAppMessageView) -> Message? {
+        getInAppMessageWebView(fromInlineView: view)?.message
     }
 
-    func onCloseActionButtonPressed() async {
+    func onCloseActionButtonPressed(onInlineView inlineView: InAppMessageView) async {
         // Triggering the close button from the web engine simulates the user tapping the close button on the in-app WebView.
         // This behaves more like an integration test because we are also able to test the message manager, too.
-        engineWebMock.delegate?.tap(name: "", action: GistMessageActions.close.rawValue, system: false)
+        getWebEngineForInlineView(inlineView)?.delegate?.tap(name: "", action: GistMessageActions.close.rawValue, system: false)
 
         // When onCloseAction() is called on the inline View, it adds a task to the main thread queue. Our test wants to wait until this task is done running.
         await waitForMainThreadToFinishPendingTasks()
     }
 
-    func isDisplayingInAppMessage(_ view: InAppMessageView) -> Bool {
-        guard let viewHeightConstraint = view.heightConstraint else {
-            return false
-        }
+    // Call when the in-app webview rendering process has finished.
+    func onDoneRenderingInAppMessage(_ message: Message, insideOfInlineView inlineView: InAppMessageView, heightOfRenderedMessage: CGFloat = 100, widthOfRenderedMessage: CGFloat = 100) async {
+        // The engine is like a HTTP layer in that it calls the Gist web server to get back rendered in-app messages.
+        // To mock the web server call with a successful response back, call these delegate functions:
+        getWebEngineForInlineView(inlineView)?.delegate?.routeLoaded(route: message.templateId)
+        getWebEngineForInlineView(inlineView)?.delegate?.sizeChanged(width: widthOfRenderedMessage, height: heightOfRenderedMessage)
 
-        return viewHeightConstraint.constant > 0
+        // When sizeChanged() is called on the inline View, it adds a task to the main thread queue. Our test wants to wait until this task is done running.
+        await waitForMainThreadToFinishPendingTasks()
+    }
+
+    func getWebEngineForInlineView(_ view: InAppMessageView) -> EngineWebInstance? {
+        view.inlineMessageManager?.engine
```

Comment:
I wanted to write test functions that involved testing multiple separate instances of inline View. In order to do that, I needed to make a refactor in the test class to have 1 `engineWebMock` instance per inline View instead of 1 mock used for all View instances. 

These utility functions were modified to take in the inline View instance, get the engine mock for that inline View, and then perform the actions on the engine instance. 

This allows us to, for example, click the close button on 1 View instance but not the other.